### ISSUE TITLE
Is certGen.sh support ecc certificate? Is certGen.sh support PoP at intermediate level?

### ISSUE THREAD

**Issue by xiongyu0523** [Azure/azure-iot-sdk-c#2011]
two questions
Is certGen.sh support ecc certificate?
Is certGen.sh support PoP at intermediate ca level? (customer want to use intermediate ca to sign leaf but do not want to put it in device)

**Comment by danewalton** [comment]
Hi @xiongyu0523 currently the script does not create ECC certs. For hints on how to create an ecc cert you can refer to our docs on our embedded SDK [here](https://github.com/Azure/azure-sdk-for-c/tree/main/sdk/samples/iot#create-a-device-using-x509-self-signed-certificate-authentication). Expand the `Instructions to create a device using X.509 Self-Signed Certificate Authentication:` and then `Certificate Generation Commands:`

As for the intermediate cert scenario, if you do not want the intermediate cert on the device, then it needs to be uploaded to DPS or IoT Hub (depending on the scenario). If you have a chain of `A -> B -> C`, the full chain needs to be available to IoT Hub at some point to verify. You can't have only A in hub and then C on the device since the chain is broken.

**Comment by xiongyu0523** [comment]
Hi, @danewalton 

Thanks. 

1. Can we bring ECC support to the script since most of beginner will use it for getting started
2. I think I can only make B (intermediate) available on IoT Hub without A, right? In this case, I need create verification certificate using private key of B, but it seems the script does not support this.. the command always use root (A) to sign the challenge

**Comment by danewalton** [comment]
ECC support in the script is not a priority at this time (as it is a preview hub feature), but we will make a note of it in our backlog to consider.

As for 2, more complex scenarios are not the intention of the script. We hope to get the basic scenarios off the ground to show how the interactions work. That being said, for unblocking purpose, there is a document [here](https://github.com/danewalton/resources/blob/master/openssl.md#ca) which may help with the various steps of openSSL commands for your scenario.
